components/server/src/ome/services/sessions/SessionManagerImpl.java

```diff
@@ -603,6 +604,28 @@ public int close(String uuid) {
         }
     }
 
+    public Map<String, EventContext> getAll() {
+        final Collection<String> ids = cache.getIds();
+        final Map<String, EventContext> rv = new HashMap<String, EventContext>();
+        for (String id : ids) {
+            if (asroot.getName().equals(id)) {
+                continue; // DON'T INCLUDE ROOT SESSION
+            }
+            try {
+                SessionContext ctx = cache.getSessionContext(id);
+                rv.put(id,  new SimpleEventContext(ctx));
+            } catch (RemovedSessionException rse) {
+                // Ok. Done for us
+            } catch (SessionTimeoutException ste) {
+                // Also ok
+            } catch (Exception e) {
+                log.warn(String.format("Exception thrown on getAll: %s:%s", e
+                        .getClass().getName(), e.getMessage()));
```

Comment:
It would warn once for each session, right? So only potentially empty, but more likely just missing the problematic sessions, I'd think.